scripts/hudson/narayana.sh

```diff
@@ -986,8 +971,13 @@ export ANT_OPTS="$ANT_OPTS $IPV6_OPTS"
 # run the job
 
 [ $NARAYANA_BUILD = 1 ] && build_narayana "$@"
-[ $AS_CLONE = 1 ] && clone_as "$@"
-[ $AS_BUILD = 1 ] && build_as "$@"
+
+if [ "$_jdk" -ge 17 ]; then
+  [ $AS_BUILD = 1 ] && AS_DOWNLOAD=1 # 26.x branch won't build on JDK 17 so download it instead
+else
+  [ $AS_CLONE = 1 ] && clone_as "$@" # why would we not set AS_BUILD=1 here
+  [ $AS_BUILD = 1 ] && build_as "$@" # should really set AS_DOWNLOAD=0 - not sure why we don't do that
```

Comment:
I think this is script is not doing what you say as it may not have been seen necessary before and we rely on the PROFILE (or environment) being set "correctly" (such as someone not introducing a profile with AS_BUILD=1 and AS_DOWNLOAD=1 or someone not exporting AS_CLONE=1 and not exporting AS_BUILD=1 at the same time as neither of those permutations would seem to be useful).

What you propose to set AS_BUILD=1 or AS_DOWNLOAD=0 would be reasonable changes I think (as would instead some earlier validation and informing the person executing the script it is not right to set inappropriate combinations). If you do change any variables for the user, please consider sending a message to console anyway to let them know what you are doing.

Reply:
I didn't follow that comment.
But I do agree that we should bail out if the script requires a WildFly build.

Reply:
I was trying to explain why it probably wasn't doing what you were asking/unsure about but re-reading I can see that is not clear - I will edit the comment. I wasn't speaking about bailing out actually

Reply:
(First comment edited now)

Reply:
[BTW in https://github.com/jbosstm/narayana/pull/2024/files I am experimenting with an option that would use AS_CLONE=1 (so we get the app server integration tests) and AS_DOWNLOAD=1 (so we wouldn't build the app server - useful for JDK 17)]